solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
+    """Sort through all *forecasts* to find those that should be
+    generated by the Arbiter from persisting Observation values. The
+    forecast must have a is_reference_persistence_forecast=true and an
+    observation_id in Forecast.extra_parameters (formatted as a JSON
+    string). A boolean value for "index_persistence" in
+    Forecast.extra_parameters controls whether the persistence
+    forecast should be made adjusting for clear-sky/AC power index or
+    not.
+
+    Parameters
+    ----------
+    session : solarforecastarbiter.io.api.APISession
+    forecasts : list of datamodel.Forecasts
+        The forecasts that should be filtered to find references.
+    observations : list of datamodel.Observations
+        Observations that will are available to use to fetch values
+        and make persistence forecasts.
+    max_run_time : pandas.Timestamp
+        The maximum run time/issue time for any forecasts. Usually now.
+
+    Returns
+    -------
+    generator of (Forecast, Observation, next_issue_time, index)
+
+    """
+    user_info = session.get_user_info()
+    observation_dict = {obs.observation_id: obs for obs in observations}
+    for fx in forecasts:
+        if not _is_reference_persistence_forecast(fx.extra_parameters):
+            logger.debug(
+                'Forecast %s is not labeled as a reference '
+                'persistence forecast',  fx.forecast_id)
+            continue
+
+        if not fx.provider == user_info['organization']:
+            logger.debug(
+                "Forecast %s is not in user's organization",
+                fx.forecast_id)
+            continue
+
+        try:
+            extra_parameters = json.loads(fx.extra_parameters)
+        except json.JSONDecodeError:
+            logger.warning(
+                'Failed to decode extra_parameters for %s: %s as JSON',
+                fx.name, fx.forecast_id)
+            continue
+
+        try:
+            observation_id = extra_parameters['observation_id']
+        except KeyError:
+            logger.error(
+                'Forecast, %s: %s, has no observation_id to base forecasts'
+                ' off of. Cannot make forecast.',
+                fx.name, fx.forecast_id)
+            continue
+        if observation_id not in observation_dict:
+            logger.error(
+                'Observation %s not in set of given observations.'
+                ' Cannot generate persistence forecast for %s:%s.',
+                observation_id, fx.name, fx.forecast_id)
+            continue
+        observation = observation_dict[observation_id]
+
+        index = extra_parameters.get('index_persistence', False)
+        obs_mint, obs_maxt = session.get_observation_time_range(observation_id)
+        if pd.isna(obs_maxt):  # no observations to use anyway
+            logger.info(
+                'No observation values to use for %s:%s from observation %s',
+                fx.name, fx.forecast_id, observation_id)
```

Comment:
consistent formatting for `'...%s:%s...', fx.name, fx.forecast_id` might make searching easier later on. The error messages above have a mix of space/no-space and sentence position.

fine to keep it this way though.